### PR TITLE
adding start/end message to timeline

### PR DESCRIPTION
**Pull request by jmervine**
addressing #154 

When incident is started without a user (Events API or push from status I'm guessing?) I've allowed this to add the timeline message anyway. I'm think catching unknown users in `timeline_command.rb` to ensure that an error isn't thrown, but rather "Unknown" is reported as the user.

I'm thinking a better way of handling this would be to create an "Events API" user, or "Status Site" user and then use that as the user when starting an incident via automation. Something like `User.create(name: "Events API", handle: nil, email: "sre-team-only@heroku.com")`. Discuss.

### CODE REVIEW

File: lib/chat_ops/commands/end_command.rb [hunk 1]
```diff
@@ -8,9 +8,12 @@ class EndCommand < ChatOps::Command
 
     def run
       @incident.chat_end = Time.zone.now
+
+      message = "Recorded the end of chat for incident `#{@incident.incident_id}` at *#{@incident.chat_end}*."
+      @incident.timeline_entries << ::TimelineEntry.new(user: @user, message: message)
```
Comment (joshuatobin) [inline lib/chat_ops/commands/end_command.rb]: 👍

**Comment by lexelby** [comment]
Looks great!  I wonder if it makes sense to allow timeline entries to have a `nil` user field instead?

**Comment by lexelby** [comment]
:+1:

**Comment by joshuatobin** [comment]
I like the idea of having an "Events" and "Status" user.

**Comment by jmervine** [comment]
Currently, they'll have a `nil` user field. Also, it's not looking like there's actually any occurrence were the "start" command is run without a user. As such, I'm going to merge this down. I do want to discuss the idea of creating canned users for events api and status site.

**Comment by reidmix** [comment]
Issue #155 btw <3

**Comment by reidmix** [comment]
I'm also a +1 on the Status user to show where it came from.  That's a fine idea.

**Pull request merged**